### PR TITLE
processor: flow control for controlling sorter output speed

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
Avoiding OOM problems

### What is changed and how it works?

block sorter output channel when a table's resolved ts is far greater than checkpoint ts to avoid existing too many events in the buffer.

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- Support a flow controler to avoid cdc server OOM

### CODE REVIEW

File: cdc/processor.go [hunk 1]
```diff
@@ -915,6 +915,8 @@ func (p *oldProcessor) addTable(ctx context.Context, tableID int64, replicaInfo
 	syncTableNumGauge.WithLabelValues(p.changefeedID, p.captureInfo.AdvertiseAddr).Inc()
 }
 
+const maxLagWithGlobalCheckpointTs = (1 * 1000) << 18 // 30s
```
Comment (leoppro) [inline cdc/processor.go:918]:
```suggestion
const maxLagWithGlobalCheckpointTs = (30 * 1000) << 18 // 30s
```

File: cdc/processor.go [hunk 2]
```diff
@@ -915,6 +915,8 @@ func (p *oldProcessor) addTable(ctx context.Context, tableID int64, replicaInfo
 	syncTableNumGauge.WithLabelValues(p.changefeedID, p.captureInfo.AdvertiseAddr).Inc()
 }
 
+const maxLagWithGlobalCheckpointTs = (30 * 1000) << 18 // 30s
```
Comment (leoppro) [inline cdc/processor.go:918]:
```suggestion
const maxLagWithGlobalCheckpointTs = (1 * 1000) << 18 // 30s
```

File: cdc/processor.go [hunk 3]
```diff
@@ -915,6 +915,8 @@ func (p *oldProcessor) addTable(ctx context.Context, tableID int64, replicaInfo
 	syncTableNumGauge.WithLabelValues(p.changefeedID, p.captureInfo.AdvertiseAddr).Inc()
 }
 
+const maxLagWithCheckpointTs = (1 * 1000) << 18 // 30s
```
Comment (leoppro) [inline cdc/processor.go:918]:
```suggestion
const maxLagWithCheckpointTs = (30 * 1000) << 18 // 30s
```

File: cdc/processor.go [hunk 4]
```diff
@@ -915,6 +915,8 @@ func (p *oldProcessor) addTable(ctx context.Context, tableID int64, replicaInfo
 	syncTableNumGauge.WithLabelValues(p.changefeedID, p.captureInfo.AdvertiseAddr).Inc()
 }
 
+const maxLagWithCheckpointTs = (30 * 1000) << 18 // 30s
```
Comment (leoppro) [inline cdc/processor.go:918]:
```suggestion
const maxLagWithCheckpointTs = (1 * 1000) << 18 // 30s
```

**Comment by amyangfei** [comment]
Is this PR ready to review

**Pull request closed without merging**